### ISSUE TITLE
Form todo

### ISSUE THREAD

**Issue by jamesplease** [jamesplease/fi-calc#285]
- [x] Replace `boolean` with `checkbox` (maybe?)
- [x] Support number and string validators
- [x] Figure out `initialValue` vs. `default`
- [x] Test the behavior of synchronous `setState` calls
- [x] Fix `"0000"` becoming `"0"`
- [x] Fix `InvestmentBubble` `onCommit`

#### Later

- [x] Move the concept of commit into app (perhaps with a `getProps` wrapper?)
- [x] Refactor outside of React
- [x] Clean up `vendor/forms`
  - [x] Restructure files
  - [x] Consider API (what should be moved out?)
  - [x] Clean up types
    - [x] Config vs Input types
- [x] Improve types of use-config-form
- [x] Rename commit-form
- [x] rename Enumeration to...Select?
- [x] ~~Consider `keyType`~~

#### Much later

- [ ] Support nested forms values `gk.upperLimit`

---

##### ✅ `initialValue` vs `default`

- `initialValue` is a `useForms` concept. It's always been there.
- `default` comes from the `useSettings` schema. I carried it over when creating the schema.

tl;dr I should refactor away `default`

**Comment by jamesplease** [comment]
This is done aside from the much-later item. Going to close and create a separate revisit issue for that one.
